sensor_core/dsp/dsp_manager.py

```diff
@@ -3,59 +3,87 @@
 import scipy.signal as signal
 
 class DSPManager(DictManager):
-    def _init_(self, dsp_module_queue: np.array([]), dsp_modules: dict = None):
+    def _init_(self, dsp_modules_queue: np.array([]), dsp_modules: dict = None):
         """Initialize empty array and create empty dictionary
         :param dsp_modules: Array of dsp_module objects, identified by user-defined name, target algo, and key parameters
         :param dsp_dict: master dsp_module dictionary
         :param dict_type: Dynamic dictionary """
 
         self.dsp_modules = {}
-        self.dsp_module_queue = []
+        self.dsp_modules_queue = []
 
+    @staticmethod
     def add_dsp_module(self, name: str, target_algo: str, **kwargs):
-        ma_keys = {'N': int, 'pad': str}
+        #TODO: Make sure no duplicate names, 
+        # Make sure input parameters are correct data type
+        # Currently only checks after run_dsp_module executes
+
+        ma_keys = {'window_size': int, 'pad': str}
         bw_keys = {'order': int, 'min_frq': int, 'max_frq': int, 'btype': str}
 
-        if (target_algo == 'moving_average_filter'):
+        if target_algo == 'moving_average_filter':
             if not set(ma_keys).issubset(kwargs.keys()):
                 raise ValueError(f"Missing required parameters: {ma_keys}")
             unexpected_params = set(kwargs.keys()) - set(ma_keys)
             if unexpected_params:
-                raise ValueError(f"Invalid moving average input, Unexpected parameters: {unexpected_params}")
+                raise ValueError(f"Invalid moving average input, unexpected parameters: {unexpected_params}")
             
             self.dsp_modules[name] = {'algo': target_algo, **kwargs}
-            self.dsp_module_queue.append(name)
+            self.dsp_modules_queue.append(name)
 
-        elif (target_algo == 'butterworth_filter'):
+        elif target_algo == 'butterworth_filter':
             if not set(bw_keys).issubset(kwargs.keys()):
                 raise ValueError(f"Missing required parameters: {bw_keys}")
             
             unexpected_params = set(kwargs.keys()) - set(bw_keys)
             if unexpected_params:
-                raise ValueError(f"Invalid moving average input, Unexpected parameters: {unexpected_params}")
+                raise ValueError(f"Invalid moving average input, unexpected parameters: {unexpected_params}")
             
             self.dsp_modules[name] = {'algo': target_algo, **kwargs}
-            self.dsp_module_queue.append(name)
+            self.dsp_modules_queue.append(name)
+
+    @staticmethod
+    def remove_dsp_module(self, name: str):
+        if name not in self.dsp_modules_queue:
+            raise ValueError(f"Specified module does not exist")
+        else:
+            rm_index = self.dsp_modules_queue.index(name)
+            self.dsp_modules_queue.pop(rm_index)
+            print(f"Update list: ", self.dsp_modules_queue)
```

Comment:
In the final version we should avoid print statements unless absolutely necessary. In this case, the user should just call the attribute `dsp_modules_queue` to see what is in the list currently